spec/02-integration/14-observability/05-logs_spec.lua

```diff
@@ -0,0 +1,86 @@
+local helpers = require "spec.helpers"
+
+for _, strategy in helpers.each_strategy() do
+  describe("Observability Logs", function ()
+    describe("ngx.log patch", function()
+      local proxy_client
+      local post_function_access = [[
+        local threads = {}
+        local n_threads = 100
+
+        for i = 1, n_threads do
+          threads[i] = ngx.thread.spawn(function()
+            ngx.log(ngx.INFO, "thread_" .. i .. " logged")
+          end)
+        end
+
+        for i = 1, n_threads do
+          ngx.thread.wait(threads[i])
+        end
+      ]]
+
+      lazy_setup(function()
+        local bp = helpers.get_db_utils(strategy, {
+          "routes",
+          "services",
+          "plugins",
+        })
+
+        local http_srv = assert(bp.services:insert {
+          name = "mock-service",
+          host = helpers.mock_upstream_host,
+          port = helpers.mock_upstream_port,
+        })
+
+        local logs_route = assert(bp.routes:insert({ service = http_srv,
+                                                     protocols = { "http" },
+                                                     paths = { "/logs" }}))
+
+        assert(bp.plugins:insert({
+          name = "post-function",
+          route = logs_route,
+          config = {
+            access = { post_function_access },
+          },
+        }))
+
+        -- only needed to enable the log collection hook
+        assert(bp.plugins:insert({
+          name = "opentelemetry",
+          route = logs_route,
+          config = {
+            logs_endpoint = "http://" .. helpers.mock_upstream_host .. ":" .. helpers.mock_upstream_port,
+          }
+        }))
+
+        helpers.start_kong({
+          database = strategy,
+          nginx_conf = "spec/fixtures/custom_nginx.template",
+          plugins = "opentelemetry,post-function",
+        })
+        proxy_client = helpers.proxy_client()
+      end)
+
+      lazy_teardown(function()
+        if proxy_client then
+          proxy_client:close()
+        end
+        helpers.stop_kong()
+      end)
+
+      it("does not produce yielding and concurrent executions", function ()
```

Comment:
Sorry, I can't understand what this test case is for, could you explain it?

Reply:
sure @ADD-SP . The test is designed to detect yielding in the `ngx.log` function by executing it concurrently and checking that [this line](https://github.com/Kong/kong/blob/b33755b694f3a4ea3dc076bef1b22389b9584e18/kong/globalpatches.lua#L601) is not logged. To make the check more strict I am looking for any `[error]` but for the sake of this check we could specifically look for something that matches `concurrent execution detected for: ngx.log`.

If ever [the hook or anything in the patched logic](https://github.com/Kong/kong/blob/b33755b694f3a4ea3dc076bef1b22389b9584e18/kong/globalpatches.lua#L612) had to yield, this test is supposed to fail.